DOCS-76 - Integrate UX style guidance (#4085)
SumoLogic/sumologic-documentation: docs/contributing/style-guide.md

* [Feathers](https://ffeathers.wordpress.com/). Blog for technical and fiction writing.
* [I'd rather be writing](https://idratherbewriting.com/). Guides and thoughts on tech writing process and content.

The Documentation team will review submissions, provide suggested edits, add new content into the navigation, and answer any questions you have.

:::tip
To create a new doc quickly, use a template. You can copy and paste the file, add your content, and submit a PR.
* [Doc template](/docs/contributing/templates/generic-doc)
* [Partner app template](/docs/contributing/templates/partner-app-doc)
:::

## Voice and tone

> * **Clarity and Professionalism**. We are clear, genuine, and trustworthy. We understand that our customers entrust us with their vital data and never make light about our commitment to data access and security.

Reviewer: Should we use sentence case for the items in this bulleted list? For example, "Clarify and professionalism", "Engagement and assistance", "Professional description", and so on?

Author: Oops, good catch! thank you.

* **Reader Address**. Address the reader as "you", as you would in conversation. For example, instead of saying, "The user must provide his or her API key" or "One must provide their API key", say, "You'll need to provide your API key".
* **Readability and SEO**. Instructional content and blog posts should be written at approximately the 8th-grade reading level, particularly in introductory sections, for readability and SEO. You can test your content [here](http://www.writingtester.com).
* **Brand Guidelines**. Our brand guideline is to always refer to “Sumo Logic”, rather than Sumo. To be more conversational, it is also fine to say “we”.
* **Gentle Language**. Use “need to” instead of “have to” or “must”. “Have to” and “must” can sound harsh and unfriendly.
* **Judicious Use of Absolutes**. Be judicious in use of “always” and “never”. Sometimes it’s appropriate to say “always” or “never”. Keep in mind though that “always” can imply a result that is not guaranteed, and “never” may very well not be the case — the exception makes the rule.

> * **Error Messaging**. When explaining a process or procedure, clarity is critical. Edit words that distract or confuse. Put yourself into the reader's shoes and think about what actions you are recommended to them when an error message is displayed, rather than merely stating what went wrong. Example: "Could not create the user." vs "This email is already registered in the system. Please use a different email or contact Sumo Logic for assistance."

Reviewer: ```suggestion
* **Error Messaging**. When explaining a process or procedure, clarity is critical. Edit words that distract or confuse. Put yourself into the reader's shoes and think about what actions you recommend to them when an error message is displayed, rather than merely stating what went wrong. Example: "Could not create the user." vs "This email is already registered in the system. Please use a different email or contact Sumo Logic for assistance."
```

Punctuation is placed outside of quotation marks, British English style. For everything else (dates, times, spelling), we use American English style.

### Colons
Colons are used to introduce lists or to separate titles from subtitles. Only include the colon if the introduction isn’t a complete sentence.

### Commas

We use the Oxford (serial) comma. For example, use "I had eggs, toast, and orange juice", not "[I had eggs, toast and orange juice](https://www.verbicidemagazine.com/wp-content/uploads/2012/01/why-i-still-use-the-oxford-comma.jpg)".

> ### Exclamation Points

Reviewer: ```suggestion
### Exclamation points
```

Use exclamation points to express excitement or encourage the user. Don't use them for errors, warnings, or confirmation of basic actions as they are usually unnecessary and can distract from important details.

### Ellipses and Truncation

Use ellipses for truncation. Don’t use ellipses for placeholders or trailing off a sentence unless it is getting cut off. When truncating, think about which part of the string is most essential for the user. This may require truncating in the beginning, middle, or end of a string.

### Periods

In the UI, avoid periods for single sentences on their own. Whenever there are two or more sentences, use periods at the end of each sentence. Separate sentences by one space, not two.

> ## Release Notes

Reviewer: ```suggestion
## Release notes
```

Reviewer: You should also change "Reusing Content" to "Reusing content".

Author: thank you - done@

* Use Google Analytics to make data-driven decisions.
:::

> ## UI Elements

Reviewer: ```suggestion
## UI elements
```

Not all UI elements have an intuitive name. It's the very nature of working with a constantly evolving product. Here are some examples:

> import UiElements from '../reuse/ui-elements.md';

Reviewer: You might want to future-proof the ui-elements.md file by using UI examples from Kanso.

Author: Thank you! I'll work on that now and resubmit.

Author: Thank you! I've created a [separate ticket](https://sumologic.atlassian.net/browse/DOCS-355) for this task. I want to run it past you guys and the UX team.
